util/chplenv/chpl_llvm.py

```diff
@@ -1024,6 +1024,42 @@ def get_clang_prgenv_args():
 
     return (comp_args, link_args)
 
+@memoize
+def get_system_include_directories(flag, lang):
```

Comment:
I'm pretty sure this code will only work for `gcc` and `clang`. I'd recommend limiting it to those compilers. We can add other compilers as needed.